In [10]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [11]:
data = pd.read_csv("temperature.csv")

data.head()

station        Date  Present_Tmax  Present_Tmin  LDAPS_RHmin  LDAPS_RHmax  \
0      1.0  30-06-2013          28.7          21.4    58.255688    91.116364   
1      2.0  30-06-2013          31.9          21.6    52.263397    90.604721   
2      3.0  30-06-2013          31.6          23.3    48.690479    83.973587   
3      4.0  30-06-2013          32.0          23.4    58.239788    96.483688   
4      5.0  30-06-2013          31.4          21.9    56.174095    90.155128   

   LDAPS_Tmax_lapse  LDAPS_Tmin_lapse  LDAPS_WS    LDAPS_LH  ...  LDAPS_PPT2  \
0         28.074101         23.006936  6.818887   69.451805  ...         0.0   
1         29.850689         24.035009  5.691890   51.937448  ...         0.0   
2         30.091292         24.565633  6.138224   20.573050  ...         0.0   
3         29.704629         23.326177  5.650050   65.727144  ...         0.0   
4         29.113934         23.486480  5.735004  107.965535  ...         0.0   

   LDAPS_PPT3  LDAPS_PPT4      lat      lon       DEM   Slope  \
0         0.0         0.0  37.6046  126.991  212.3350  2.7850   
1         0.0         0.0  37.6046  127.032   44.7624  0.5141   
2         0.0         0.0  37.5776  127.058   33.3068  0.2661   
3         0.0         0.0  37.6450  127.022   45.7160  2.5348   
4         0.0         0.0  37.5507  127.135   35.0380  0.5055   

   Solar radiation  Next_Tmax  Next_Tmin  
0      5992.895996       29.1       21.2  
1      5869.312500       30.5       22.5  
2      5863.555664       31.1       23.9  
3      5856.964844       31.7       24.3  
4      5859.552246       31.2       22.5  

[5 rows x 25 columns]

In [12]:
data.shape

(7752, 25)

Our dataset contains 7752 rows (observations) and 25 columns(features).

In [13]:
data.dtypes

station             float64
Date                 object
Present_Tmax        float64
Present_Tmin        float64
LDAPS_RHmin         float64
LDAPS_RHmax         float64
LDAPS_Tmax_lapse    float64
LDAPS_Tmin_lapse    float64
LDAPS_WS            float64
LDAPS_LH            float64
LDAPS_CC1           float64
LDAPS_CC2           float64
LDAPS_CC3           float64
LDAPS_CC4           float64
LDAPS_PPT1          float64
LDAPS_PPT2          float64
LDAPS_PPT3          float64
LDAPS_PPT4          float64
lat                 float64
lon                 float64
DEM                 float64
Slope               float64
Solar radiation     float64
Next_Tmax           float64
Next_Tmin           float64
dtype: object

As the "Date" column has in object data type we need to convert it into numeric data type.

In [14]:
data["Date"] = pd.to_datetime(data.Date, format='%d-%m-%Y')

In [15]:
data.reset_index(inplace = True)
data["Year"] = data["Date"].dt.year
data["Month"] = data["Date"].dt.month
data["Day"] = data["Date"].dt.day

data.set_index("Date",inplace = True)

In [16]:
data.head()

index  station  Present_Tmax  Present_Tmin  LDAPS_RHmin  \
Date                                                                  
2013-06-30      0      1.0          28.7          21.4    58.255688   
2013-06-30      1      2.0          31.9          21.6    52.263397   
2013-06-30      2      3.0          31.6          23.3    48.690479   
2013-06-30      3      4.0          32.0          23.4    58.239788   
2013-06-30      4      5.0          31.4          21.9    56.174095   

            LDAPS_RHmax  LDAPS_Tmax_lapse  LDAPS_Tmin_lapse  LDAPS_WS  \
Date                                                                    
2013-06-30    91.116364         28.074101         23.006936  6.818887   
2013-06-30    90.604721         29.850689         24.035009  5.691890   
2013-06-30    83.973587         30.091292         24.565633  6.138224   
2013-06-30    96.483688         29.704629         23.326177  5.650050   
2013-06-30    90.155128         29.113934         23.486480  5.735004   

              LDAPS_LH  ...      lat      lon       DEM   Slope  \
Date                    ...                                       
2013-06-30   69.451805  ...  37.6046  126.991  212.3350  2.7850   
2013-06-30   51.937448  ...  37.6046  127.032   44.7624  0.5141   
2013-06-30   20.573050  ...  37.5776  127.058   33.3068  0.2661   
2013-06-30   65.727144  ...  37.6450  127.022   45.7160  2.5348   
2013-06-30  107.965535  ...  37.5507  127.135   35.0380  0.5055   

            Solar radiation  Next_Tmax  Next_Tmin    Year  Month   Day  
Date                                                                    
2013-06-30      5992.895996       29.1       21.2  2013.0    6.0  30.0  
2013-06-30      5869.312500       30.5       22.5  2013.0    6.0  30.0  
2013-06-30      5863.555664       31.1       23.9  2013.0    6.0  30.0  
2013-06-30      5856.964844       31.7       24.3  2013.0    6.0  30.0  
2013-06-30      5859.552246       31.2       22.5  2013.0    6.0  30.0  

[5 rows x 28 columns]

In [17]:
data.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 7752 entries, 2013-06-30 to NaT
Data columns (total 28 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   index             7752 non-null   int64  
 1   station           7750 non-null   float64
 2   Present_Tmax      7682 non-null   float64
 3   Present_Tmin      7682 non-null   float64
 4   LDAPS_RHmin       7677 non-null   float64
 5   LDAPS_RHmax       7677 non-null   float64
 6   LDAPS_Tmax_lapse  7677 non-null   float64
 7   LDAPS_Tmin_lapse  7677 non-null   float64
 8   LDAPS_WS          7677 non-null   float64
 9   LDAPS_LH          7677 non-null   float64
 10  LDAPS_CC1         7677 non-null   float64
 11  LDAPS_CC2         7677 non-null   float64
 12  LDAPS_CC3         7677 non-null   float64
 13  LDAPS_CC4         7677 non-null   float64
 14  LDAPS_PPT1        7677 non-null   float64
 15  LDAPS_PPT2        7677 non-null   float64
 16  LDAPS_PPT3        7677 non-null

In [18]:
data.describe()

index      station  Present_Tmax  Present_Tmin  LDAPS_RHmin  \
count  7752.000000  7750.000000   7682.000000   7682.000000  7677.000000   
mean   3875.500000    13.000000     29.768211     23.225059    56.759372   
std    2237.953976     7.211568      2.969999      2.413961    14.668111   
min       0.000000     1.000000     20.000000     11.300000    19.794666   
25%    1937.750000     7.000000     27.800000     21.700000    45.963543   
50%    3875.500000    13.000000     29.900000     23.400000    55.039024   
75%    5813.250000    19.000000     32.000000     24.900000    67.190056   
max    7751.000000    25.000000     37.600000     29.900000    98.524734   

       LDAPS_RHmax  LDAPS_Tmax_lapse  LDAPS_Tmin_lapse     LDAPS_WS  \
count  7677.000000       7677.000000       7677.000000  7677.000000   
mean     88.374804         29.613447         23.512589     7.097875   
std       7.192004          2.947191          2.345347     2.183836   
min      58.936283         17.624954         14.272646     2.882580   
25%      84.222862         27.673499         22.089739     5.678705   
50%      89.793480         29.703426         23.760199     6.547470   
75%      93.743629         31.710450         25.152909     8.032276   
max     100.000153         38.542255         29.619342    21.857621   

          LDAPS_LH  ...          lat          lon          DEM        Slope  \
count  7677.000000  ...  7752.000000  7752.000000  7752.000000  7752.000000   
mean     62.505019  ...    37.544722   126.991397    61.867972     1.257048   
std      33.730589  ...     0.050352     0.079435    54.279780     1.370444   
min     -13.603212  ...    37.456200   126.826000    12.370000     0.098475   
25%      37.266753  ...    37.510200   126.937000    28.700000     0.271300   
50%      56.865482  ...    37.550700   126.995000    45.716000     0.618000   
75%      84.223616  ...    37.577600   127.042000    59.832400     1.767800   
max     213.414006  ...    37.645000   127.135000   212.335000     5.178230   

       Solar radiation    Next_Tmax    Next_Tmin         Year        Month  \
count      7752.000000  7725.000000  7725.000000  7750.000000  7750.000000   
mean       5341.502803    30.274887    22.932220  2015.000000     7.467742   
std         429.158867     3.128010     2.487613     1.414305     0.530334   
min        4329.520508    17.400000    11.300000  2013.000000     6.000000   
25%        4999.018555    28.200000    21.300000  2014.000000     7.000000   
50%        5436.345215    30.500000    23.100000  2015.000000     7.000000   
75%        5728.316406    32.600000    24.600000  2016.000000     8.000000   
max        5992.895996    38.900000    29.800000  2017.000000     8.000000   

               Day  
count  7750.000000  
mean     15.983871  
std       8.918647  
min       1.000000  
25%       8.000000  
50%      16.000000  
75%      24.000000  
max      31.000000  

[8 rows x 28 columns]

Checking null values.

In [19]:
data.isnull().sum()

index                0
station              2
Present_Tmax        70
Present_Tmin        70
LDAPS_RHmin         75
LDAPS_RHmax         75
LDAPS_Tmax_lapse    75
LDAPS_Tmin_lapse    75
LDAPS_WS            75
LDAPS_LH            75
LDAPS_CC1           75
LDAPS_CC2           75
LDAPS_CC3           75
LDAPS_CC4           75
LDAPS_PPT1          75
LDAPS_PPT2          75
LDAPS_PPT3          75
LDAPS_PPT4          75
lat                  0
lon                  0
DEM                  0
Slope                0
Solar radiation      0
Next_Tmax           27
Next_Tmin           27
Year                 2
Month                2
Day                  2
dtype: int64

As we seen the data the last 2 rows don't have station and date info so we will drop.

In [20]:
data.drop(data.tail(2).index, inplace = True)

In [21]:
data.isnull().sum()

index                0
station              0
Present_Tmax        70
Present_Tmin        70
LDAPS_RHmin         75
LDAPS_RHmax         75
LDAPS_Tmax_lapse    75
LDAPS_Tmin_lapse    75
LDAPS_WS            75
LDAPS_LH            75
LDAPS_CC1           75
LDAPS_CC2           75
LDAPS_CC3           75
LDAPS_CC4           75
LDAPS_PPT1          75
LDAPS_PPT2          75
LDAPS_PPT3          75
LDAPS_PPT4          75
lat                  0
lon                  0
DEM                  0
Slope                0
Solar radiation      0
Next_Tmax           27
Next_Tmin           27
Year                 0
Month                0
Day                  0
dtype: int64

Now station and date doesn't contain any null values so we will fill the remaining null values.

Gone through the dataset we observed that all the columns are grouped in 3 group.Based on the number of missing values now we will check whether if these missing values are at the same indices. 

In [29]:
index_present_tmax = data['Present_Tmax'].index[data['Present_Tmax'].apply(np.isnan)]
index_present_tmin = data['Present_Tmin'].index[data['Present_Tmin'].apply(np.isnan)]

index_next_tmax = data['Next_Tmax'].index[data['Next_Tmax'].apply(np.isnan)]
index_next_tmin = data['Next_Tmin'].index[data['Next_Tmin'].apply(np.isnan)]

index_LDAPS_RHmax = data['LDAPS_RHmax'].index[data['LDAPS_RHmax'].apply(np.isnan)]
index_LDAPS_RHmin = data['LDAPS_RHmin'].index[data['LDAPS_RHmin'].apply(np.isnan)]
index_LDAPS_Tmax_lapse = data['LDAPS_Tmax_lapse'].index[data['LDAPS_Tmax_lapse'].apply(np.isnan)]
index_LDAPS_Tmin_lapse = data['LDAPS_Tmin_lapse'].index[data['LDAPS_Tmin_lapse'].apply(np.isnan)]
index_LDAPS_WS = data['LDAPS_WS'].index[data['LDAPS_WS'].apply(np.isnan)]
index_LDAPS_LH = data['LDAPS_LH'].index[data['LDAPS_LH'].apply(np.isnan)]
index_LDAPS_CC1 = data['LDAPS_CC1'].index[data['LDAPS_CC1'].apply(np.isnan)]
index_LDAPS_CC2 = data['LDAPS_CC2'].index[data['LDAPS_CC2'].apply(np.isnan)]
index_LDAPS_CC3 = data['LDAPS_CC3'].index[data['LDAPS_CC3'].apply(np.isnan)]
index_LDAPS_CC4 = data['LDAPS_CC4'].index[data['LDAPS_CC4'].apply(np.isnan)]
index_LDAPS_PPT1 = data['LDAPS_PPT1'].index[data['LDAPS_PPT1'].apply(np.isnan)]
index_LDAPS_PPT2 = data['LDAPS_PPT2'].index[data['LDAPS_PPT2'].apply(np.isnan)]
index_LDAPS_PPT3 = data['LDAPS_PPT3'].index[data['LDAPS_PPT3'].apply(np.isnan)]
index_LDAPS_PPT4 = data['LDAPS_PPT4'].index[data['LDAPS_PPT4'].apply(np.isnan)]



if list (index_present_tmax) == list(index_present_tmin):
    print('Lists are identical')
else: 
    print('Unidentical lists')
    
if list (index_next_tmax) == list(index_next_tmin):
    print('Lists are identical')
else:
    print('Unidentical lists')
    
    
if list(index_LDAPS_RHmax) == list(index_LDAPS_RHmin) == list(index_LDAPS_Tmax_lapse) == list(index_LDAPS_Tmin_lapse) == list(index_LDAPS_WS) == list(index_LDAPS_LH) == list(index_LDAPS_CC1) == list(index_LDAPS_CC2) == list(index_LDAPS_CC3) == list(index_LDAPS_CC4) == list(index_LDAPS_PPT1) == list(index_LDAPS_PPT2) == list(index_LDAPS_PPT3) == list(index_LDAPS_PPT4):
    print('Lists are identical')
else:
    print('Unidentical lists')

Lists are identical
Lists are identical
Lists are identical
